### ISSUE TITLE
`doctest` repeats ignored test on macOS

### ISSUE THREAD

**Issue by ProfFan** [rust-lang/rust#127893]
I tried this code:

```rust
/// ```no_run
///  panic!("nooo");
///  ```
pub struct XXXXYYYY<A, B> {
...
}
```

I expected to see this happen: does not run

Instead, this happened:

```
running 2 tests
test src/lib.rs - XXXXYYYY (line 23) - compile ... ok
test src/lib.rs - (line 23) ... FAILED
```

### Meta

`rustc --version --verbose`:
```
rustc 1.81.0-nightly (032be6f7b 2024-07-16)
binary: rustc
commit-hash: 032be6f7bbe091c7dfa29f115e94b9cc9bae1758
commit-date: 2024-07-16
host: aarch64-apple-darwin
release: 1.81.0-nightly
LLVM version: 18.1.7
```

<details><summary>Backtrace</summary>
<p>

```
N/A
```

</p>
</details>

**Comment by ProfFan** [comment]
Strange, this runs even after I removed the doctest itself, `target` folder, as well as doing a `cargo clean`. Does `cargo` keep a cache of this test somewhere outside of the project?

**Comment by workingjubilee** [comment]
...weird.

can you provide a version that compiles and repros?

**Comment by ProfFan** [comment]
> ...weird.
> 
> can you provide a version that compiles and repros?

https://github.com/ProfFan/icm426xx on `main`

I don't know if this is my macOS being funky, this is the only system I have.

Just run `cargo test` and this should trigger?

**Comment by ProfFan** [comment]
nvm, it's this trick from stackoverflow that got me:
```
#![doc = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/README.md"))]
```

This results in doctest trying to run the code snippet from README.

However this problem is still valid: the displayed line number is clearly incorrect. It shows the line number of my disabled test in the source code.

**Comment by ProfFan** [comment]
XREF: https://stackoverflow.com/questions/69117817/how-can-i-tell-cargo-to-run-code-test-in-documentation-in-my-readme-md

Same problem in this post.

**Comment by tgross35** [comment]
Similar but not exact issue https://github.com/rust-lang/rust/issues/81070
